src/lib.ts

```diff
@@ -138,50 +141,89 @@ function acceptBestGuzzlrQuest() {
+  if ($location`The Skeleton Store`.turnsSpent >= 3)
+    backupBlacklist.push($location`The Skeleton Store`);
+  if ($location`The Overgrown Lot`.turnsSpent >= 5)
+    backupBlacklist.push($location`The Overgrown Lot`);
+
+  const canBackup = have($item`backup camera`) && get("_backUpUses") < 11;
+
   if (!Guzzlr.have()) return defaultLocation;
 
   acceptBestGuzzlrQuest();
-  if (!guzzlrCheck()) Guzzlr.abandon();
+
+  if (
+    !guzzlrCheck() ||
+    (type === draggableFight.WANDERER &&
+      !canBackup &&
+      wandererBlacklist.includes(Guzzlr.getLocation() || $location`none`)) ||
+    (type === draggableFight.BACKUP &&
+      !canWander() &&
+      backupBlacklist.includes(Guzzlr.getLocation() || $location`none`) &&
+      (Guzzlr.getLocation()?.combatPercent || 100) < 100)
+  ) {
+    Guzzlr.abandon(); //Abandon if it's generically bad, or if it's bad for the only type of draggable fight you have left
+  }
+
   acceptBestGuzzlrQuest();
 
   const guzzlZone = Guzzlr.getLocation();
   if (!guzzlrCheck()) return defaultLocation;
-  else if (!guzzlZone) return defaultLocation;
-  else {
-    if (Guzzlr.getTier() === "platinum") {
-      zonePotions.forEach((place) => {
-        if (guzzlZone.zone === place.zone && !have(place.effect)) {
-          if (!have(place.potion)) {
-            buy(1, place.potion, 10000);
-          }
-          use(1, place.potion);
+  if (!guzzlZone) return defaultLocation;
+
+  if (
+    type === draggableFight.BACKUP &&
+    (guzzlZone.combatPercent < 100 || backupBlacklist.includes(guzzlZone))
+  ) {
+    print("We're backing up, and our guzzlr zone doesn't like that.", "red");
+    return defaultLocation;
+  }
+  if (type === draggableFight.WANDERER && wandererBlacklist.includes(guzzlZone)) {
+    print("We're dropping a wanderer, and our guzzlr zone doesn't like that.", "red");
+    return defaultLocation;
+  }
+  if (Guzzlr.getTier() === "platinum") {
+    zonePotions.forEach((place) => {
+      if (guzzlZone.zone === place.zone && !have(place.effect)) {
+        if (!have(place.potion)) {
+          buy(1, place.potion, 10000);
         }
-      });
-      if (!Guzzlr.havePlatinumBooze()) {
-        cliExecute("make buttery boy");
-      }
-    } else {
-      const guzzlrBooze = Guzzlr.getBooze();
-      if (!guzzlrBooze) {
-        return defaultLocation;
-      } else if (!have(guzzlrBooze)) {
-        print(`just picking up some booze before we roll`, "blue");
-        cliExecute(`acquire ${guzzlrBooze}`);
+        use(1, place.potion);
       }
+    });
+    if (!Guzzlr.havePlatinumBooze()) {
+      create($item`Buttery Boy`);
+    }
+  } else {
+    const guzzlrBooze = Guzzlr.getBooze();
+    if (!guzzlrBooze) {
+      return defaultLocation;
+    } else if (!have(guzzlrBooze)) {
+      print(`just picking up some booze before we roll`, "blue");
+      retrieveItem(guzzlrBooze);
     }
-    return guzzlZone;
   }
+  return guzzlZone;
 }
 
-function guzzlrCheck() {
+function guzzlrCheck(): boolean {
```

Comment:
not sure what this does either per the name

Reply:
renamed to checkGuzzlrZone. Is that more clear? The intent is that this tells us if we categorically hate a particular guzzlr zone. It does not reject zones that we only hate for wanderers or that we only hate for backups.

Reply:
I thought about doWeHateOurGuzzlrZone but thought that might be too fanciful. But that's the next option in line for names if checkGuzzlrZone is still unclear